tests/core/desktop-files/task.yaml

```diff
@@ -0,0 +1,6 @@
+summary: Ensure that desktop files installed on Core systems
+
+execute: |
+    echo "Desktop files are present on Ubuntu Core systems"
+    test -f /var/lib/snapd/desktop/applications/io.snapcraft.SessionAgent.desktop
+    test -f /var/lib/snapd/desktop/applications/snap-handle-link.desktop
```

Comment:
This should test whether the files exist in `/var/lib/snapd/desktop/applications` or `/usr/share/applications`. As is, I suspect this will fail on the ubuntu-core-16-* backends.

The `wrappers/core18.go` code is not run for Core 16: there, the snapd files are directly integrated into the base snap.

Alternatively, you could exclude the test from `ubuntu-core-16-*`.